lib/deprecated-locales.json

```diff
@@ -1,5 +1,6 @@
 {
     "iw": "he",
     "ji": "yi",
-    "in": "id"
+    "in": "id",
+    "cmn": "zh"
```

Comment:
If anything, it’s `zh` that’s deprecated, not `cmn`, so maybe we should rename this file to something more generic, like language-aliases.json.

Reply:
This takes care of falling back from `cmn` to `zh`, which is necessary in some cases, but the client might need to fall back in the other direction, if the locale to match is `zh` but only `cmn` or `yue` is available.

Reply:
I have some other opinions for the other direction.
`cmn` should be defined as a language code but not a locale. I think the locale should be exclusive but there can be different language codes. The strategy I think should be, we can use a language code to find out a best locale, and using a locale to find out a best language code. 
But now our `locale-utils` seems combine `locale` and `language code` together.

Reply:
By locale, I mean [language tag](https://en.wikipedia.org/wiki/IETF_language_tag), not ISO region code. If a tool that uses this library has `en` and `cmn` as available locale codes, then an input of `zh`, `zh-CN`, or `zh-US` should probably match to `cmn` rather than `en`.

Reply:
Make sense. But there will be a issue, when the user both use `cmn` and `zh` for available locales, there will be a conflict.

Reply:
When both `cmn` and `zh-Hans` are available, then no fallback is used, and the result depends on whether `inputLocale` starts with `cmn` or `zh`.